Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -15,129 +15,113 @@ import TorusUtils
             throw CoreKitError.notFound(msg: "share index not found")
         }
         return shareIndex
     }
 
-    // TODO: This should accept a param class instead
-    public func loginWithOAuth(loginProvider: LoginProviders, clientId: String, verifier: String, jwtParams: [String: String] = [:], redirectURL: String = "tdsdk://tdsdk/oauthCallback", browserRedirectURL: String = "https://scripts.toruswallet.io/redirect.html") async throws -> MpcKeyDetails {
-        if loginProvider == .jwt && jwtParams.isEmpty {
-            throw CoreKitError.notFound(msg: "jwt login should provide jwtParams")
-        }
+    public func loginWithOAuth(singleLoginParams: SingleLoginParams) async throws -> MpcKeyDetails {
+        let loginResponse = try await customAuth.triggerLogin(args: singleLoginParams)
 
-        let sub = SubVerifierDetails(loginType: .web,
-                                     loginProvider: loginProvider,
-                                     clientId: clientId,
-                                     verifier: verifier,
-                                     redirectURL: redirectURL,
-                                     browserRedirectURL: browserRedirectURL,
-                                     jwtParams: jwtParams
-        )
-        
-        let customAuth = CustomAuth(web3AuthClientId: option.Web3AuthClientId, aggregateVerifierType: .singleLogin, aggregateVerifier: verifier, subVerifierDetails: [sub], network: network.toTorusNetwork(), enableOneKey: true)
+        let result = try await login(keyDetails: loginResponse.torusKey, verifier: loginResponse.singleVerifierResponse.userInfo.verifier, verifierId: loginResponse.singleVerifierResponse.userInfo.verifierId)
+        userInfo = loginResponse.singleVerifierResponse.userInfo
 
-        let userData = try await customAuth.triggerLogin()
-        return try await login(userData: userData)
+        return result
+    }
+
+    public func loginWithOAuth(aggregateLoginParams: AggregateLoginParams) async throws -> MpcKeyDetails {
+        let loginResponse = try await customAuth.triggerAggregateLogin(args: aggregateLoginParams)
+        let result = try await login(keyDetails: loginResponse.torusKey, verifier: loginResponse.torusAggregateVerifierResponse.first!.userInfo.verifier, verifierId: loginResponse.torusAggregateVerifierResponse.first!.userInfo.verifierId)
+
+        userInfo = loginResponse.torusAggregateVerifierResponse.first?.userInfo
+
+        return result
     }
 
-    // mneomonic to share
     public func mnemonicToKey(shareMnemonic: String, format: String) throws -> String {
-        // Assuming ShareSerializationModule.deserializeMnemonic returns Data
         let factorKey = try ShareSerializationModule.deserialize_share(threshold_key: tkey!, share: shareMnemonic, format: format)
         return factorKey
     }
 
-    // share to mneomonic
     public func keyToMnemonic(factorKey: String, format: String) throws -> String {
```

Comment:
format should have proper type for values it can accept.

Reply:
@AyushBherwani1998

Reply:
After discussing with @ieow we decided to remove the the format, since the accepted parameter is either `nil` or `mnemonic`.